src/test/java/org/zalando/nakadi/controller/EventStreamControllerTest.java

```diff
@@ -466,6 +472,24 @@ public void testAccessDenied() throws Exception {
         assertThat(responseToString(responseBody), TestUtils.JSON_TEST_HELPER.matchesObject(expectedProblem));
     }
 
+    @Test
+    public void testAccessAllowedForAllDataAccess() throws Exception {
```

Comment:
This class test Low Level API stream. What about a test for Subscription API?

Reply:
@v-stepanov Sorry, I'm not going to write a test for Subscription API. The reason is:
- to write a unit test means a nightmare of mocks (I spent too much time trying, it's still not working and there are a lot of mocks already)
- ideally we should have an acceptance test. I think this should be written elegantly when we have an endpoint to set applications that can read from all. I could populate the database with some values on startup... but that's also ugly.

What do you think?

Reply:
@lmontrieux 
Yes, probably it will be difficult to write AT checking authorization because we run ATs without any tokens. And unit test will have too much to mock. Ok, let's do it without a test.